pkg/agent-functions/agents/agents.go

```diff
@@ -110,7 +111,8 @@ func (a Agents) GetAgents() ([]string, error) {
 
 	agents := []string{}
 	for _, kv := range resp.Kvs {
-		agents = append(agents, re.ReplaceAllString(string(kv.Key), ""))
+		agent := re.ReplaceAllString(string(kv.Key), "")
```

Comment:
@sahil-lakhwani,

It is too fragile to rely on the key format. It will break any time the key format is changed in the future.

Instead decode the value as [Peer protobuf message](https://github.com/fluxninja/aperture/blob/main/api/aperture/peers/v1/peers.proto#L49)

Reply:
Also, do not [hardcode etcd paths](https://github.com/fluxninja/aperture/pull/2815/files#diff-54eaa7e7c0cc107d41212dd60f644b9b3edb6616ae9918239ead654f94cb95bcR102). We have these defined in a common place as constants

Reply:
Modified to use proto structs.
Declared the etcd path as const in same file